hamilton/include/hamilton/hybrid_planner.h

```diff
@@ -0,0 +1,133 @@
+                                                                     /* Typically "world_frame" or "base_link". */
+    std::vector<std::string> joint_names;                            /* A list with the names of the mobile joints in the robot */ 
+                                                                     // 2nd param of toROSJointTrajectory()
+    // MoveIt! params
+    double plan_time;   // = 10.0;                      
+    double step_size;  // = 0.01; 
+    double jump_thresh; // =0;
+    bool moveit_replan;
+    bool avoid_collision;
+    std::string moveit_planner_type;
+    
+    // Descartes params
+    std::string descartes_planner_type;
+    double time_delay; // = 1.0; Last param of toROSJointTrajectory() 
+    double time_offset; // = 0.0; Used inside toROSJointTrajectory()
+  };
+
+  /*  =============================== TrajectorySegment Struct ===============================
+  * Struct to store the unplanned segments, and categorize them as free and process path. 
+  * Note: The third type "FREE_MOTION_START_END_ONLY" isn't used right now.
+  * The idea is to call the MoveIt planner via a MotionPlanning Request in this case, via the PlanHybrid method.  
+  * TODO Add more data structures in TrajectorySegment - joint vals, descartes type itself, etc */
+
+  struct TrajectorySegment
+  {
+    TrajectorySegmentType type_;      
+    std::vector<geometry_msgs::Pose> waypoints_geom_msgs_;
+  }; 
+
+  /*  =============================== Hybrid Planner Class ===============================
+   * Provides a group of functions for planning a robot path using Moveit and  the Descartes Planning Library */
+
+  class HybridPlanner
+  {
+    public:
+      HybridPlanner();
+      virtual ~HybridPlanner();
+      // Initialize (descartes) robot model and planners
+      void init();
+
+    protected:
+      // Append unplanned free space/process path segments defined by "waypoints' to unplanned_trajectory_ as a TrajectorySegment object. 
+      // TODO extend to other types - joints, descartes types, etc.
+    	void appendFreeMotionTrajectorySegment(std::vector<geometry_msgs::Pose>& waypoints);
+    	void appendProcessPathTrajectorySegment(std::vector<geometry_msgs::Pose>& waypoints);
+
+      // Appends unplanned free space segment, defined by start and end point only as a TrajectorySegment object. 
+      void appendFreeMotionTrajectorySegment(geometry_msgs::Pose start_pose, geometry_msgs::Pose end_pose);
+
+      // Iterates through the unplanned_trajectory_vector, and calls MoveIt! and Descartes planners as required.
+      // Saves the planned, joint space trajectory in overall_robot_traj_, which can be send for execution to the controller.
+      // TODO Not Implemented: Makes a motion planning request if the TrajetorySegemnt.type_ == FREE_MOTION_START_END_ONLY
+    	void PlanHybrid();
```

Comment:
This method name doesn't follow the [method naming convention](http://wiki.ros.org/CppStyleGuide#Function_.2BAC8_Methods)